packages/core/frontend/sirius-components-omnibox/src/Omnibox.tsx

```diff
@@ -0,0 +1,179 @@
+    } = event;
+    setState((prevState) => ({ ...prevState, query: value }));
+  };
+
+  useEffect(() => {
+    const variables: GQLGetOmniboxCommandsQueryVariables = {
+      contextEntries: state.contextEntries,
+      query: state.query,
+    };
+    getOmniboxCommands({ variables });
+  }, [state.query]);
+
+  useEffect(() => {
+    if (!loading && data) {
+      const actions: OmniboxAction[] = data.viewer.omniboxCommands.edges
+        .map((edge) => edge.node)
+        .map((node) => {
+          return {
+            id: node.id,
+            icon: <IconOverlay iconURL={node.iconURL} alt={node.kind} />,
+            kind: node.kind,
+            label: node.label,
+          };
+        });
+      setState((prevState) => ({ ...prevState, actions }));
+    }
+  }, [loading, data]);
+
+  const handleKeyDown: React.KeyboardEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
+    if (event.key === 'Enter') {
+      onClose();
+    } else if (event.key === 'ArrowDown' && listRef.current) {
+      const firstListItem = listRef.current.childNodes[0];
+      if (firstListItem instanceof HTMLElement) {
+        firstListItem.focus();
+      }
+    }
+  };
+
+  const handleListItemKeyDown: React.KeyboardEventHandler<HTMLDivElement> = (event) => {
+    if (event.key === 'ArrowDown') {
+      const nextListItemButton = event.currentTarget.nextSibling;
+      if (nextListItemButton instanceof HTMLElement) {
+        nextListItemButton.focus();
+      }
+    } else if (event.key === 'ArrowUp') {
+      const previousListItemButton = event.currentTarget.previousSibling;
+      if (previousListItemButton instanceof HTMLElement) {
+        previousListItemButton.focus();
+      }
+    } else {
+      const input = inputRef.current?.querySelector('input');
+      if (input) {
+        input.focus();
+      }
+    }
+  };
+
+  const handleOnActionClick = (action: OmniboxAction) => {
+    setSelection({ entries: [{ id: action.id, kind: action.kind }] });
+    onClose();
+  };
+
+  const { classes } = useOmniboxStyles();
+  return (
+    <Dialog open={open} onClose={onClose} fullWidth className={classes.omnibox}>
+      <div className={classes.omniboxInputArea}>
+        <ChevronRightIcon className={classes.omniboxIcon} />
+        <Breadcrumbs>
+          {state.contextEntries.map((contextEntry) => {
+            return <Typography key={contextEntry.id}>{contextEntry.label}</Typography>;
+          })}
+        </Breadcrumbs>
+        <FormControl variant="standard" fullWidth className={classes.omniboxFormControl}>
+          <Input
+            ref={inputRef}
+            value={state.query}
+            onChange={handleChange}
+            onKeyDown={handleKeyDown}
+            placeholder="Run commands..."
+            disableUnderline
+            autoFocus
+            fullWidth
+            inputProps={{
+              style: {
+                fontSize: '1.5rem',
+              },
+            }}
+          />
+        </FormControl>
```

Comment:
Please add an Enter icon at the end of the input area and replace it by a loading one when we are waiting for the results